pkg/windowsdriver/procmon/procmon.go

```diff
@@ -148,23 +140,41 @@ func (wp *WinProcmon) Start() error {
 }
 
 //nolint:revive // TODO(WKIT) Fix revive linter
-func decodeStruct(data []uint8, sz uint32) (t DDProcessNotifyType, pid uint64, imagefile, cmdline string, consumed uint32) {
+func decodeStruct(data []uint8, sz uint32) (start *ProcessStartNotification, stop *ProcessStopNotification, consumed uint32) {
+	start = nil
+	stop = nil
 	if unsafe.Sizeof(DDProcessNotification{}.Size) > uintptr(sz) {
-		return
+		return nil, nil, 0
 	}
 
 	n := *(*DDProcessNotification)(unsafe.Pointer(&data[0]))
 	if n.Size > uint64(sz) {
-		return
+		return nil, nil, 0
 	}
 
 	consumed = uint32(n.Size)
-	pid = uint64(n.ProcessId)
-	t = DDProcessNotifyType(n.NotifyType)
+	t := DDProcessNotifyType(n.NotifyType)
 
 	if t == ProcmonNotifyStart {
-		imagefile = winutil.ConvertWindowsString(data[n.ImageFileOffset : n.ImageFileOffset+n.ImageFileLen])
-		cmdline = winutil.ConvertWindowsString(data[n.CommandLineOffset : n.CommandLineOffset+n.CommandLineLen])
+		imagefile := winutil.ConvertWindowsString(data[n.ImageFileOffset : n.ImageFileOffset+n.ImageFileLen])
+		cmdline := winutil.ConvertWindowsString(data[n.CommandLineOffset : n.CommandLineOffset+n.CommandLineLen])
+		var sidstring string
+		if n.SidLen > 0 {
+			sidstring = winutil.ConvertWindowsString(data[n.SidOffset : n.SidOffset+n.SidLen])
+		}
+		start = &ProcessStartNotification{
+			Pid:               n.ProcessId,
+			PPid:              n.ParentProcessId,
+			CreatingProcessId: n.CreatingProcessId,
+			CreatingThreadId:  n.CreatingThreadId,
+			ImageFile:         imagefile,
+			CmdLine:           cmdline,
+			OwnerSidString:    sidstring,
```

Comment:
Kind of related to prior comment about the length check - if `ImageFile`, `CmdLine`, `OwnerSidString` are empty after the conversion, will the assignment here "just work" and give us an empty string (`""`)?